### ISSUE TITLE
Problems writing MPC programs using eigen,osqp,osqp-eigen in VS2015

### ISSUE THREAD

**Issue by DRETQ** [robotology/osqp-eigen#169]
Can anyone tell me what the problem is?
![捕获](https://github.com/user-attachments/assets/78e71d46-9896-4d5e-ba24-280bffbddcca)

**Comment by S-Dafarra** [comment]
Hi @DRETQ, can you describe:
- The version of osqp-eigen and osqp you used
- The command and the compiler you used for compiling the project?
Thanks!

**Comment by DRETQ** [comment]
@S-Dafarra The version of osqp-eigen is 0.8.1, the version of osqp is 0.6.3，I used the debug x86 local windows debugger command and the compiler is cmake.

**Comment by S-Dafarra** [comment]
Given the error message, the problem seems similar to https://github.com/robotology/osqp-eigen/issues/162#issuecomment-2135097303

> Are you 100% sure you did not accidentally used osqp master branch? Because this error is exactly what you get if you are compiling against osqp master on Windows, see https://github.com/robotology/osqp-eigen/issues/146 and https://github.com/osqp/osqp/issues/576 for the actual issue tracking the problem.

Also, can you report the ``CMakeCache.txt`` of both ``osqp`` and ``osqp-eigen`` to understand if something is going wrong?

One possible attempt is to try using the ``master`` version of ``osqp-eigen`` given that https://github.com/robotology/osqp-eigen/pull/163 should have fixed the issue.

One final remark, please avoid pasting screenshots of errors, but rather copy and paste them. This would allow others to find the same solution by searching for the same error online, and would allow me to translate errors more easily.

**Comment by DRETQ** [comment]
[CMakeCache(osqp).txt](https://github.com/user-attachments/files/16265950/CMakeCache.osqp.txt)
[CMakeCache(osqp-eigen).txt](https://github.com/user-attachments/files/16265952/CMakeCache.osqp-eigen.txt)
Thanks for your reply, what is the master version of osqp-eigen? I apologize for any inconvenience caused by the use of screenshots.

**Comment by S-Dafarra** [comment]
Could you do 
``
cd E:/LibraryVS15/osqp-master/osqp
git log -1
``
and post the output?




> Thanks for your reply, what is the master version of osqp-eigen?

You can do
``
cd E:/LibraryVS15/osqp-eigen-master
git checkout master
git pull
``
and then compile again

**Comment by DRETQ** [comment]
@S-Dafarra 
![22](https://github.com/user-attachments/assets/70184c3c-27bf-400f-9fe7-26795a14bf58)
![23](https://github.com/user-attachments/assets/5ca5bee0-cf78-4619-ae6c-cae6776882b5)

**Comment by S-Dafarra** [comment]
How did you clone (download) the repo?

**Comment by DRETQ** [comment]
@S-Dafarra Do you mean how do I download osqp,osqp-eigen, I run the command from cmd to download online

**Comment by S-Dafarra** [comment]
> I run the command from cmd to download online

Just to understand, which command?

From https://github.com/robotology/osqp-eigen/issues/169#issuecomment-2235219983 it seems that you don't have ``git`` installed

**Comment by DRETQ** [comment]
@S-Dafarra Commands used when installing osqp is :git clone --recursive -b release-0.6.3 https://github.com/oxfordcontrol/osqp.git.  Configuring eigen,osqp,osqp-eigen on vs2015 should be successful because the official example file MPCExample given is able to run successfully.

**Comment by S-Dafarra** [comment]
Ok, then it seems you have ``git`` installed somewhere. Using the same application you used todo ``git clone``, do also the commands I mentioned in https://github.com/robotology/osqp-eigen/issues/169#issuecomment-2233781833

**Comment by DRETQ** [comment]
@S-Dafarra You mean you want me to use git clone instead of git. Here are the results.
![24](https://github.com/user-attachments/assets/24680971-ef1f-4fd1-9f16-295cac677454)
![25](https://github.com/user-attachments/assets/ef75bf72-61cf-4578-9bf2-b9c27ae21f21)

**Comment by DRETQ** [comment]
@S-Dafarra I found inconsistencies in reporting errors on vs2015 and vs2022，the osqp,eigen,osqp-eigen configured in it are the same.

**Comment by S-Dafarra** [comment]
> @S-Dafarra You mean you want me to use git clone instead of git.

No, I meant to run the commands I posted above in a terminal where git can be used

**Comment by DRETQ** [comment]
@S-Dafarra Where can I run git if cmd doesn't work?

**Comment by S-Dafarra** [comment]
Apologies, I misunderstood the error message of https://github.com/robotology/osqp-eigen/issues/169#issuecomment-2235219983

The problem was not that ``git`` was not working. The problem was that the system did not go to the directory where you have ``osqp-eigen``. Can you navigate to the folder where you have cloned ``osqp-eigen`` and do
```
git checkout master
git pull
```
?

**Comment by DRETQ** [comment]
Where i have cloned osqp-eigen is E:\LibraryVS15\osqp-eigen-master,isn't it? I don't know what you mean.

**Comment by S-Dafarra** [comment]
> Where i have cloned osqp-eigen is E:\LibraryVS15\osqp-eigen-master,isn't it? I don't know what you mean.

Yes, please navigate to that folder and from within that folder execute the commands
```
git checkout master
git pull
```

**Comment by DRETQ** [comment]
@S-Dafarra  Isn't that what I did before, with the same output as above?

**Comment by traversaro** [comment]
In command prompt `cd` only works inside the active drive. So if the current working directory is `C:\Windows\system32` if you do:

~~~
cd E:\LibraryVS15\osqp-eigen-master
~~~

the current working directory will remain in `C:\Windows\system32`, as the active drive remains `C:`.

To go to `E:\LibraryVS15\osqp-eigen-master`, you need to first make `E:` the active drive by typing `E:`, and then call `cd E:\LibraryVS15\osqp-eigen-master`, i.e. : 

~~~
C:\Windows\system32>E:
E:>cd E:\LibraryVS15\osqp-eigen-master
E:\LibraryVS15\osqp-eigen-master>
~~~

**Comment by DRETQ** [comment]
@S-Dafarra What you mean is that I need to do git log -1, git checkout master, and git pull in osqp and osqp-eigen folders respectively, not in windows system32.
![26](https://github.com/user-attachments/assets/e81b9846-8f96-4295-853f-d8a173a59900)
![27](https://github.com/user-attachments/assets/4af17796-15a7-4ff5-aadb-0557230b4e6c)

**Comment by DRETQ** [comment]
@traversaro Thanks for the reply, I've remembered

**Comment by S-Dafarra** [comment]
Where is the source code of ``osqp-eigen``? Maybe ``E:/LibraryVS15/osqp-eigen-master/osqp-eigen-0.8.1``?

**Comment by DRETQ** [comment]
@S-Dafarra 
![28](https://github.com/user-attachments/assets/858cfb74-fdc5-4b52-ac41-de6e921f5ae7)

**Comment by S-Dafarra** [comment]
Neither ``E:/LibraryVS15/osqp-eigen-master/osqp-eigen-0.8.1`` is a ``git`` repository. Where did you clone the repo?

**Comment by DRETQ** [comment]
@S-Dafarra When installing osqp-eigen, I didn't choose to use the git hub command directly with the osqp install, but instead downloaded the optional installer from the web.

**Comment by S-Dafarra** [comment]
> @S-Dafarra When installing osqp-eigen, I didn't choose to use the git hub command directly with the osqp install, but instead downloaded the optional installer from the web.

Ok then use ``git clone`` to download the repo and try switching to the ``master`` branch.

**Comment by DRETQ** [comment]
@S-Dafarra This is a problem with debug x86 when debugging, it doesn't seem to be a problem if you use debug x64

**Comment by S-Dafarra** [comment]
> @S-Dafarra This is a problem with debug x86 when debugging, it doesn't seem to be a problem if you use debug x64

Ah good catch. To be honest I don't think we test in CI x86 builds, let alone VS2015

**Comment by DRETQ** [comment]
@S-Dafarra Yes, I've asked before and no one has used it on VS2015. And have a new question, I have defined the sparse hessian matrix as a square matrix in my program, why does the output say this：
osqpeigen::data::sethessianmatrix the hessian matrix has to be a n*n size
![image](https://github.com/user-attachments/assets/8d327539-cea4-45ac-bef5-782f0dd9f462)

**Comment by traversaro** [comment]
Hello @DRETQ, please:
* Use separate issues for each question (otherwise it is quite difficult for anyone to read issue if we mix different problems)
* If you have a problem on a code, please report the code you are compiling and running, and the complete error, ideally in text form instead of copy&pasting images.

Thanks!

**Comment by traversaro** [comment]
I think we can close here given that VS2015 is not supported anymore in osqp-eigen.
